components/collector/src/source_collectors/gitlab/pipeline_duration.py

```diff
@@ -0,0 +1,22 @@
+"""GitLab CI-pipeline duration collector."""
+
+from collector_utilities.type import URL, Value
+from model import SourceResponses
+
+from .base import GitLabPipelineBase
+
+
+class GitLabPipelineDuration(GitLabPipelineBase):
+    """GitLab CI-pipeline duration collector."""
+
+    async def _get_source_responses(self, *urls: URL) -> SourceResponses:
+        """Override to get the pipeline details."""
+        responses = await super()._get_source_responses(*urls)
+        pipelines = await self._pipelines(responses)
+        api_url = await self._gitlab_api_url(f"pipelines/{pipelines[0].id}")
+        return await super()._get_source_responses(api_url)
```

Comment:
Do I understand it correctly to summarise that we "get all pipelines", "select the first" and then "return that pipeline data"? How then do we select the highest from X recent pipeline runs?

Reply:
And does this logic not conflict with other Gitlab pipeline metrics we offer, which consider individual stages/jobs?

Reply:
As mentioned above, this metric measures the duration of the latest pipeline.